### PR TITLE
Add TTL enforcement

### PR DESCRIPTION
**Pull request by chliang71**
### What changes are proposed in this pull request?

This change adds a scheduled thread in LocalCacheManager, to periodically check the page create time, and delete pages that have passed the configurable time threshold.

This change is on top of this PR: https://github.com/Alluxio/alluxio/pull/16783/files

### Why are the changes needed?

Certain use cases require the cached data to be removed beyond certain longevity. This is mainly for compliance considerations.

### Does this PR introduce any user facing changes?

No

### CODE REVIEW

**Review by beinan — COMMENTED** [review]
It looks good to me,  @ChunxuTang would you like to take a look also?

File: core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java [hunk 1]
```diff
@@ -141,6 +148,21 @@ public static LocalCacheManager create(CacheManagerOptions options,
...
+          return System.currentTimeMillis() - pageInfo.getCreatedTimestamp() >=
+                  options.getTtlThresholdSeconds() * 1000;
+        } catch (Exception ex) {
+          // In case of any exception, do not invalidate the cache
+          return false;
+        }
+      }), 0, options.getTtlCheckIntervalSeconds(), SECONDS);
+    } else {
+      mTtlEnforcerExecutor = Optional.empty();
+    }
```
Comment (ChunxuTang) [inline core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java:165]: The indentation here looks a bit weird. Would you mind double-checking whether this fulfills the coding style by running "mvn checkstyle:checkstyle"?

File: core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java [hunk 2]
```diff
@@ -88,6 +93,8 @@ public class LocalCacheManager implements CacheManager {
   private final Optional<ExecutorService> mInitService;
   /** Executor service for execute the async cache tasks. */
   private final Optional<ExecutorService> mAsyncCacheExecutor;
+  /** Executor service for execute the cache ttl check tasks. */
+  private final Optional<ScheduledExecutorService> mTtlEnforcerExecutor;
```
Comment (ChunxuTang) [inline core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java:97]: Have you tested the new feature in the `LocalCacheManager`? I noticed that there's a corresponding unit test file `LocalCacheManagerTest.java`. Do you want to add some tests there?

**Review by ChunxuTang — COMMENTED** [review]
@chliang71 Thanks for your contribution!

I just left some minor suggestions. Also, would you mind merging your latest commit "Fix checkstyle" with the previous commit "Cache TTL enforcement"?

**Comment by beinan** [comment]
> 



> @chliang71 Thanks for your contribution!
> 
> I just left some minor suggestions. Also, would you mind merging your latest commit "Fix checkstyle" with the previous commit "Cache TTL enforcement"?

No worry, alluxio's git bot will squash all the commits into one when we merge it.  So feel free to keep the commits

**Comment by ChunxuTang** [comment]
Hi @chliang71 and @beinan, as we've merged the PR https://github.com/Alluxio/alluxio/pull/16843 and Uber is testing this feature, I'm closing this PR. Feel free to reopen it if necessary.

**Pull request closed without merging**